ops/ccs-ops-misc/synthetic-data/scripts/synthea-automation/prepare-and-run-synthea.py

```diff
@@ -251,53 +265,40 @@ def check_ranges(properties_file, number_of_benes_to_generate, db_string):
-    query = f"select count(*) from hospice_claims where fi_doc_clm_cntl_num::bigint <= {fi_num_start}"
-    result = _execute_single_count_query(db_string, query)
-    overall_validation_result = field_has_room_in_table(fi_num_start, "hospice_claims", "fi_doc_clm_cntl_num", result, overall_validation_result)
-    
+    overall_validation_result = overall_validation_result and check_single_doesnt_exist('outpatient_claims', 'fi_doc_clm_cntl_num', fi_num_start, db_string)
+    overall_validation_result = overall_validation_result and check_single_doesnt_exist('inpatient_claims', 'fi_doc_clm_cntl_num', fi_num_start, db_string)
+    overall_validation_result = overall_validation_result and check_single_doesnt_exist('hha_claims', 'fi_doc_clm_cntl_num', fi_num_start, db_string)
+    overall_validation_result = overall_validation_result and check_single_doesnt_exist('snf_claims', 'fi_doc_clm_cntl_num', fi_num_start, db_string)
+    overall_validation_result = overall_validation_result and check_single_doesnt_exist('hospice_claims', 'fi_doc_clm_cntl_num', fi_num_start, db_string)
+
     ## Since MBI and HICN are incremented in difficult-to-query ways, just check if it exists; if it doesnt exist the range should be fine
     hicn_start = get_props_value(properties_file, "exporter.bfd.hicn_start")
-    query = f"select count(*) from beneficiaries where hicn_unhashed = \'{hicn_start}\'"
-    result = _execute_single_count_query(db_string, query)
-    if result > 0:
-        print(f"(Validation Failure) Start HICN {hicn_start} exists in DB")
-        overall_validation_result = False
-    else:
-        print("(Validation Success) Start HICN does not exist in DB")
+    overall_validation_result = overall_validation_result and check_single_doesnt_exist('beneficiaries', 'hicn_unhashed', hicn_start, db_string)
         
     mbi_start = get_props_value(properties_file, "exporter.bfd.mbi_start")
-    query = f"select count(*) from beneficiaries where mbi_num = \'{mbi_start}\'"
-    result = _execute_single_count_query(db_string, query)
-    if result > 0:
-        print(f"(Validation Failure) Start MBI {mbi_start} exists in DB")
-        overall_validation_result = False
-    else:
-        print("(Validation Success) Start MBI does not exist in DB")
+    overall_validation_result = overall_validation_result and check_single_doesnt_exist('beneficiaries', 'mbi_num', mbi_start, db_string)
         
-    ## Ensure no mbi_hash has more than one bene_id associated with it from any previous loads
-    query = "select count(*) from (select mbi_hash, bene_id, count(bene_id) from beneficiaries_history where mbi_hash IS NOT NULL group by mbi_hash, bene_id having count(bene_id) > 1) as s"
+    ## Ensure no (synthetic) mbi_num has more than one bene_id associated with it from any previous loads
+    ## this takes about 30 seconds in test as of this writing, and 2 seconds in the other envs, but this will increase as more data is added
```

Comment:
If this timing is a concern, there are probably some pretty easy to implement indexing strategies that could help out. Something like:
```sql
CREATE INDEX my_index_name ON public.beneficiaries (bene_id, mbi_num)
WHERE bene_id < 0 AND mbi_num IS NOT NULL;
```

Reply:
(With that index, the queries below might not even hit the actual table data, so that may be an added bonus?)

Reply:
Will have to see how it scales once we have more data; for now 30 seconds isnt a huge time to wait for this step. If we start getting into waiting for minutes (and we may eventually) putting an index like this would be a good idea.